src/main/scala/is/hail/methods/LinearRegression.scala

```diff
@@ -19,7 +19,11 @@ object LinearRegression {
     require(vds.wasSplit)
 
     val (y, cov, completeSamples) = RegressionUtils.getPhenoCovCompleteSamples(vds, yExpr, covExpr)
-    val sampleMask = vds.sampleIds.map(completeSamples.toSet).toArray
+    val completeSamplesSet = completeSamples.toSet
```

Comment:
consider encapsulating lines 22-25 in a function on completeSamples in RegressionUtils